{"name": "Тест: переход по клику на «Конструктор»", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.NoSuchWindowException: Message: no such window: target window already closed\nfrom unknown error: web view not found\n  (Session info: chrome=131.0.6778.205)\nStacktrace:\n\tGetHandleVerifier [0x00007FF60E81FB05+28789]\n\t(No symbol) [0x00007FF60E7886E0]\n\t(No symbol) [0x00007FF60E62592A]\n\t(No symbol) [0x00007FF60E5FF505]\n\t(No symbol) [0x00007FF60E6A6477]\n\t(No symbol) [0x00007FF60E6BEF42]\n\t(No symbol) [0x00007FF60E69F1E3]\n\t(No symbol) [0x00007FF60E66A938]\n\t(No symbol) [0x00007FF60E66BAA1]\n\tGetHandleVerifier [0x00007FF60EB5933D+3410093]\n\tGetHandleVerifier [0x00007FF60EB6E7DD+3497293]\n\tGetHandleVerifier [0x00007FF60EB62A73+3448803]\n\tGetHandleVerifier [0x00007FF60E8E7BBB+848171]\n\t(No symbol) [0x00007FF60E793C3F]\n\t(No symbol) [0x00007FF60E78F6E4]\n\t(No symbol) [0x00007FF60E78F87D]\n\t(No symbol) [0x00007FF60E77ED49]\n\tBaseThreadInitThunk [0x00007FFD25D07374+20]\n\tRtlUserThreadStart [0x00007FFD25F5CC91+33]", "trace": "self = <main_functionality_test.TestMainFunctional object at 0x000001FA04C95810>, driver = <selenium.webdriver.chrome.webdriver.WebDriver (session=\"cc7490a6dc61f6c66d6bee133f181301\")>\nmain_func = <pages.main_page.MainPage object at 0x000001FA04D18050>\n\n    @allure.title(\"Тест: переход по клику на «Конструктор»\")\n    def test_click_on_constructor(self, driver, main_func):\n>       main_func.click_on_personal_account_button()\n\ntests\\main_functionality_test.py:16: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\npages\\base_page.py:17: in click_on_personal_account_button\n    self.driver.find_element(*main_page_personal_account).click()\n..\\..\\..\\AppData\\Local\\Programs\\Python\\Python313\\Lib\\site-packages\\selenium\\webdriver\\remote\\webdriver.py:748: in find_element\n    return self.execute(Command.FIND_ELEMENT, {\"using\": by, \"value\": value})[\"value\"]\n..\\..\\..\\AppData\\Local\\Programs\\Python\\Python313\\Lib\\site-packages\\selenium\\webdriver\\remote\\webdriver.py:354: in execute\n    self.error_handler.check_response(response)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <selenium.webdriver.remote.errorhandler.ErrorHandler object at 0x000001FA04D19550>\nresponse = {'status': 404, 'value': '{\"value\":{\"error\":\"no such window\",\"message\":\"no such window: target window already closed\\\\...07FF60E77ED49]\\\\n\\\\tBaseThreadInitThunk [0x00007FFD25D07374+20]\\\\n\\\\tRtlUserThreadStart [0x00007FFD25F5CC91+33]\\\\n\"}}'}\n\n    def check_response(self, response: Dict[str, Any]) -> None:\n        \"\"\"Checks that a JSON response from the WebDriver does not have an\n        error.\n    \n        :Args:\n         - response - The JSON response from the WebDriver server as a dictionary\n           object.\n    \n        :Raises: If the response contains an error message.\n        \"\"\"\n        status = response.get(\"status\", None)\n        if not status or status == ErrorCode.SUCCESS:\n            return\n        value = None\n        message = response.get(\"message\", \"\")\n        screen: str = response.get(\"screen\", \"\")\n        stacktrace = None\n        if isinstance(status, int):\n            value_json = response.get(\"value\", None)\n            if value_json and isinstance(value_json, str):\n                import json\n    \n                try:\n                    value = json.loads(value_json)\n                    if len(value) == 1:\n                        value = value[\"value\"]\n                    status = value.get(\"error\", None)\n                    if not status:\n                        status = value.get(\"status\", ErrorCode.UNKNOWN_ERROR)\n                        message = value.get(\"value\") or value.get(\"message\")\n                        if not isinstance(message, str):\n                            value = message\n                            message = message.get(\"message\")\n                    else:\n                        message = value.get(\"message\", None)\n                except ValueError:\n                    pass\n    \n        exception_class: Type[WebDriverException]\n        e = ErrorCode()\n        error_codes = [item for item in dir(e) if not item.startswith(\"__\")]\n        for error_code in error_codes:\n            error_info = getattr(ErrorCode, error_code)\n            if isinstance(error_info, list) and status in error_info:\n                exception_class = getattr(ExceptionMapping, error_code, WebDriverException)\n                break\n        else:\n            exception_class = WebDriverException\n    \n        if not value:\n            value = response[\"value\"]\n        if isinstance(value, str):\n            raise exception_class(value)\n        if message == \"\" and \"message\" in value:\n            message = value[\"message\"]\n    \n        screen = None  # type: ignore[assignment]\n        if \"screen\" in value:\n            screen = value[\"screen\"]\n    \n        stacktrace = None\n        st_value = value.get(\"stackTrace\") or value.get(\"stacktrace\")\n        if st_value:\n            if isinstance(st_value, str):\n                stacktrace = st_value.split(\"\\n\")\n            else:\n                stacktrace = []\n                try:\n                    for frame in st_value:\n                        line = frame.get(\"lineNumber\", \"\")\n                        file = frame.get(\"fileName\", \"<anonymous>\")\n                        if line:\n                            file = f\"{file}:{line}\"\n                        meth = frame.get(\"methodName\", \"<anonymous>\")\n                        if \"className\" in frame:\n                            meth = f\"{frame['className']}.{meth}\"\n                        msg = \"    at %s (%s)\"\n                        msg = msg % (meth, file)\n                        stacktrace.append(msg)\n                except TypeError:\n                    pass\n        if exception_class == UnexpectedAlertPresentException:\n            alert_text = None\n            if \"data\" in value:\n                alert_text = value[\"data\"].get(\"text\")\n            elif \"alert\" in value:\n                alert_text = value[\"alert\"].get(\"text\")\n            raise exception_class(message, screen, stacktrace, alert_text)  # type: ignore[call-arg]  # mypy is not smart enough here\n>       raise exception_class(message, screen, stacktrace)\nE       selenium.common.exceptions.NoSuchWindowException: Message: no such window: target window already closed\nE       from unknown error: web view not found\nE         (Session info: chrome=131.0.6778.205)\nE       Stacktrace:\nE       \tGetHandleVerifier [0x00007FF60E81FB05+28789]\nE       \t(No symbol) [0x00007FF60E7886E0]\nE       \t(No symbol) [0x00007FF60E62592A]\nE       \t(No symbol) [0x00007FF60E5FF505]\nE       \t(No symbol) [0x00007FF60E6A6477]\nE       \t(No symbol) [0x00007FF60E6BEF42]\nE       \t(No symbol) [0x00007FF60E69F1E3]\nE       \t(No symbol) [0x00007FF60E66A938]\nE       \t(No symbol) [0x00007FF60E66BAA1]\nE       \tGetHandleVerifier [0x00007FF60EB5933D+3410093]\nE       \tGetHandleVerifier [0x00007FF60EB6E7DD+3497293]\nE       \tGetHandleVerifier [0x00007FF60EB62A73+3448803]\nE       \tGetHandleVerifier [0x00007FF60E8E7BBB+848171]\nE       \t(No symbol) [0x00007FF60E793C3F]\nE       \t(No symbol) [0x00007FF60E78F6E4]\nE       \t(No symbol) [0x00007FF60E78F87D]\nE       \t(No symbol) [0x00007FF60E77ED49]\nE       \tBaseThreadInitThunk [0x00007FFD25D07374+20]\nE       \tRtlUserThreadStart [0x00007FFD25F5CC91+33]\n\n..\\..\\..\\AppData\\Local\\Programs\\Python\\Python313\\Lib\\site-packages\\selenium\\webdriver\\remote\\errorhandler.py:229: NoSuchWindowException"}, "steps": [{"name": "Клик по кнопке Личный кабинет", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.NoSuchWindowException: Message: no such window: target window already closed\nfrom unknown error: web view not found\n  (Session info: chrome=131.0.6778.205)\nStacktrace:\n\tGetHandleVerifier [0x00007FF60E81FB05+28789]\n\t(No symbol) [0x00007FF60E7886E0]\n\t(No symbol) [0x00007FF60E62592A]\n\t(No symbol) [0x00007FF60E5FF505]\n\t(No symbol) [0x00007FF60E6A6477]\n\t(No symbol) [0x00007FF60E6BEF42]\n\t(No symbol) [0x00007FF60E69F1E3]\n\t(No symbol) [0x00007FF60E66A938]\n\t(No symbol) [0x00007FF60E66BAA1]\n\tGetHandleVerifier [0x00007FF60EB5933D+3410093]\n\tGetHandleVerifier [0x00007FF60EB6E7DD+3497293]\n\tGetHandleVerifier [0x00007FF60EB62A73+3448803]\n\tGetHandleVerifier [0x00007FF60E8E7BBB+848171]\n\t(No symbol) [0x00007FF60E793C3F]\n\t(No symbol) [0x00007FF60E78F6E4]\n\t(No symbol) [0x00007FF60E78F87D]\n\t(No symbol) [0x00007FF60E77ED49]\n\tBaseThreadInitThunk [0x00007FFD25D07374+20]\n\tRtlUserThreadStart [0x00007FFD25F5CC91+33]\n\n", "trace": "  File \"C:\\Users\\mmavr\\AppData\\Local\\Programs\\Python\\Python313\\Lib\\site-packages\\allure_commons\\_allure.py\", line 192, in impl\n    return func(*a, **kw)\n  File \"C:\\Users\\mmavr\\yandex\\Diplom\\Diplom_3\\pages\\base_page.py\", line 17, in click_on_personal_account_button\n    self.driver.find_element(*main_page_personal_account).click()\n    ~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\mmavr\\AppData\\Local\\Programs\\Python\\Python313\\Lib\\site-packages\\selenium\\webdriver\\remote\\webdriver.py\", line 748, in find_element\n    return self.execute(Command.FIND_ELEMENT, {\"using\": by, \"value\": value})[\"value\"]\n           ~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\mmavr\\AppData\\Local\\Programs\\Python\\Python313\\Lib\\site-packages\\selenium\\webdriver\\remote\\webdriver.py\", line 354, in execute\n    self.error_handler.check_response(response)\n    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^\n  File \"C:\\Users\\mmavr\\AppData\\Local\\Programs\\Python\\Python313\\Lib\\site-packages\\selenium\\webdriver\\remote\\errorhandler.py\", line 229, in check_response\n    raise exception_class(message, screen, stacktrace)\n"}, "start": 1734734135575, "stop": 1734734135576}], "start": 1734734135575, "stop": 1734734135578, "uuid": "27052390-c8f5-4fec-828d-21f8d67efbee", "historyId": "2434cb8ca9916e7c5cbb2209f5507fe3", "testCaseId": "2434cb8ca9916e7c5cbb2209f5507fe3", "fullName": "tests.main_functionality_test.TestMainFunctional#test_click_on_constructor", "labels": [{"name": "tag", "value": "@pytest.mark.usefixtures('main_func')"}, {"name": "parentSuite", "value": "tests"}, {"name": "suite", "value": "main_functionality_test"}, {"name": "subSuite", "value": "TestMainFunctional"}, {"name": "host", "value": "DESKTOP-7LU8FN0"}, {"name": "thread", "value": "10216-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "tests.main_functionality_test"}]}